cmd/bosun/expr/funcs.go

```diff
@@ -691,11 +692,16 @@ func timeTSDBRequest(e *State, T miniprofiler.Timer, req *opentsdb.Request) (s o
 	return
 }
 
-func timeLSRequest(e *State, T miniprofiler.Timer, service *elastic.SearchService, source *elastic.SearchSource) (resp *elastic.SearchResult, err error) {
-	e.logstashQueries = append(e.logstashQueries, *service.SearchSource(source))
-	b, _ := json.MarshalIndent(source.Source(), "", "  ")
+func timeLSRequest(e *State, T miniprofiler.Timer, req *logstash.Request) (resp *elastic.SearchResult, err error) {
+	e.logstashQueries = append(e.logstashQueries, *req.Source)
+	b, _ := json.MarshalIndent(req.Source.Source(), "", "  ")
 	T.StepCustomTiming("logstash", "query", string(b), func() {
-		resp, err = service.SearchSource(source).Do()
+		getFn := func() (interface{}, error) {
+			return e.logstashContext.Query(req)
+		}
+		var val interface{}
+		val, err = e.cache.Get(string(b), getFn)
```

Comment:
cache stores pointer to data. but the whole expression package was made with the decision that changing data is ok, so when avg() modifies the result it's working on, we don't want that to affect the data in cache, so we must copy the data as we pull it out of cache.  i'm not sure where we agreed that should be done, i think the caller of the cache (the query functions in funcs.go) have to take care of this, but @captncraig probably remembers this better?